src/wallet/rpc/wallet.cpp

```diff
@@ -745,6 +745,10 @@ static RPCHelpMan migratewallet()
                 throw JSONRPCError(RPC_WALLET_WRONG_ENC_STATE, "Error: migratewallet on encrypted wallets is currently unsupported.");
             }
 
+            if (!wallet->GetLegacyScriptPubKeyMan()) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Error: nothing to migrate, no legacy data contained in wallet \"%s\"", wallet->GetName()));
```

Comment:
I wonder if the message is consistent with how the migration code actually works. For instance, the error returned before this change is "`This wallet already uses SQLite`":

https://github.com/bitcoin/bitcoin/blob/fbe5e1220a35c35a668ec965c66508a7839aa7dd/src/wallet/wallet.cpp#L3765-L3774

Even if it passed that check (which I believe is impossible in this situation), it does the same check as in this PR and reads "`This wallet is already a descriptor wallet`":

https://github.com/bitcoin/bitcoin/blob/fbe5e1220a35c35a668ec965c66508a7839aa7dd/src/wallet/wallet.cpp#L3837-L3845


Minor additional nit: I think the RPC error messages typically start with a capital letter after the `Error: ` unless it's an RPC command (as in line 745 above).

Reply:
> I wonder if the message is consistent with how the migration code actually works. For instance, the error returned before this change is "This wallet already uses SQLite"

well, think about the command purpose. If there is no legacy data to migrate then why the command should be executed at all? Doesn't really care if the db is sqlite, bdb or anything else.

> Minor additional nit: I think the RPC error messages typically start with a capital letter after the Error:  unless it's an RPC command (as in line 745 above).

Seems that we have a mix of them, some errors start with a capital letter, some others not (like the encrypt.cpp file). I'm more aligned to this second form but have no real preference for any. Maybe the best move would be to open an issue an align all of them, in one way or another, at once.

Reply:
> If there is no legacy data to migrate then why the command should be executed at all?

Right, I agree that the message makes logical sense, though I do worry that it suggests that someone can have a descriptor wallet with legacy data, or they may think that "legacy" addresses are "legacy data", like `getnewaddress -addresstype legacy`.

My personal preference is to copy the `Error: This wallet is already a descriptor wallet` message, since that better matches the `help migratewallet` message: `Migrate the wallet to a descriptor wallet.` 

That said, it's not a strong preference. Maybe @achow101 could weigh in?

Reply:
> Right, I agree that the message makes logical sense, though I do worry that it suggests that someone can have a descriptor wallet with legacy data

It's theoretically possible to have a descriptor wallet using BDB for the database as well as a legacy wallet using sqlite. These are unsupported configurations, but are allowed and possible to achieve using the wallettool. That being said, they are unsupported so I think it's fine, and the number of people who would have such wallets should be limited to a very small number of developers.

Regarding the error message, I would also prefer the "already a descriptor wallet" wording.

Reply:
pushed.
M moved the `GetDescriptorsForLegacy` error up to the top of the process to not duplicate it.